lib/Dancer2/Core/Role/Config.pm

```diff
@@ -107,62 +116,45 @@ sub _build_environment { 'development' }
 sub _build_config_files {
     my ($self) = @_;
 
-    my $location = $self->config_location;
-    # an undef location means no config files for the caller
-    return [] unless defined $location;
-
-    my $running_env = $self->environment;
-    my @available_exts = Config::Any->extensions;
-    my @files;
-
-    my @exts = @available_exts;
-    if (my $ext = $ENV{DANCER_CONFIG_EXT}) {
-        if (grep { $ext eq $_ } @available_exts) {
-            @exts = $ext;
-            warn "Only looking for configs ending in '$ext'\n" 
-                if $ENV{DANCER_CONFIG_VERBOSE};
-        } else {
-            warn "DANCER_CONFIG_EXT environment variable set to '$ext' which\n" .
-                 "is not recognized by Config::Any. Looking for config file\n" .
-                 "using default list of extensions:\n" .
-                 "\t@available_exts\n";
-        }
-    }
-
-    foreach my $file ( [ $location, "config" ],
-        [ $self->environments_location, $running_env ] )
-    {
-        foreach my $ext (@exts) {
-            my $path = path( $file->[0], $file->[1] . ".$ext" );
-            next if !-r $path;
-
-            # Look for *_local.ext files
-            my $local = path( $file->[0], $file->[1] . "_local.$ext" );
-            push @files, $path, ( -r $local ? $local : () );
-        }
-    }
-
-    return \@files;
+    return [ map {
+            warn "Merging config_files from @{[ $_->name() ]}\n" if $ENV{DANCER_CONFIG_VERBOSE};
+            @{ $_->config_files() }
+        } @{ $self->config_readers }
+    ];
 }
 
+# The new config builder
 sub _build_config {
     my ($self) = @_;
 
-    my $location = $self->config_location;
     my $default  = $self->default_config;
-
     my $config = Hash::Merge::Simple->merge(
         $default,
         map {
-            warn "Merging config file $_\n" if $ENV{DANCER_CONFIG_VERBOSE};
-            $self->load_config_file($_) 
-        } @{ $self->config_files }
+            warn "Merging config from @{[ $_->name() ]}\n" if $ENV{DANCER_CONFIG_VERBOSE};
+            $_->read_config()
+        } @{ $self->config_readers }
     );
 
     $config = $self->_normalize_config($config);
     return $config;
 }
 
+sub _build_config_readers {
+    my ($self) = @_;
+
+    my @config_reader_names = $ENV{'DANCER_CONFIG_READERS'}
+                            ? (split qr{ [[:space:]]{1,} }msx, $ENV{'DANCER_CONFIG_READERS'})
```

Comment:
Is there a reason the `split` is written this way with POSIX character classes instead of `split /\s+/xms, $ENV{'DANCER_CONFIG_READERS'}` ?

Reply:
POSIX removed, replaced with `\s+`.